ecosystem/indexer-grpc/indexer-transaction-generator/src/transaction_generator.rs

```diff
@@ -76,12 +97,185 @@ impl TransactionGenerator {
         self.managed_node.stop().await
     }
 
+    /// Go through the test case steps and generate the transactions.
     async fn generate(&self, test_case: TestCase) -> anyhow::Result<()> {
-        // TODO: finish the test generation logic.
         println!(
-            "Generate transaction for test case {} at {:?} ",
+            "\n Generate transaction for test case {} at {:?} \n",
             test_case.name, self.output_folder,
         );
+        // Test case output folder.
+        let test_case_output_folder = self.output_folder.join(&test_case.name);
+        // If the output folder doesn't exist, create it.
+        if !test_case_output_folder.exists() {
+            std::fs::create_dir_all(&test_case_output_folder)?;
+        }
+        let mut transactions_version_to_capture = vec![];
+        // Execute the transactions.
+        for step in test_case.steps {
```

Comment:
Rather than access `test_case.steps` directly shouldn't there be a `steps()` function that returns the steps sorted by priority?